src/org/etools/j1939_84/controllers/part2/Part02Step13Controller.java

```diff
@@ -49,6 +62,38 @@
 
     @Override
     protected void run() throws Throwable {
-       getListener().onResult("There are " + dataRepository.getObdModuleAddresses() + " OBD Modules");
+        //6.2.13 DM31: DTC to Lamp Association
+        // 6.2.13.1 Actions:
+        //  a. DS DM31 (send Request (PGN 59904) for PGN 41728 (SPNs 1214-1215, 4113, 4117)) to each OBD ECU.
+        List<DM31DtcToLampAssociation> dsPackets = new ArrayList<>();
+        List<AcknowledgmentPacket> ackPackets = new ArrayList<>();
+        dataRepository.getObdModules().forEach(module -> {
+            RequestResult<DM31DtcToLampAssociation> requestResult = getDiagnosticMessageModule()
+                    .requestDM31(getListener(), module.getSourceAddress());
+            if(!requestResult.getPackets().isEmpty()){
+                dsPackets.addAll(requestResult.getPackets());
+            }
+            if(!requestResult.getAcks().isEmpty()){
+                ackPackets.addAll(requestResult.getAcks());
+            }
+        });
+
+        // 6.2.13.2 Fail criteria (if supported):
+        dsPackets.stream()
+                .filter(packet -> isMilNotOffAndNotAltOff(packet))
+                .forEach( packet -> {
+            addFailure("6.2.13.2.a - ECU " + Lookup.getAddressName(packet.getSourceAddress())
+                               + " reported MIL light not off/alt-off");
```

Comment:
Nit.  MIL is “Malfunction Indicator Lamp”. MIL Light is redundant.  “ATM Machine”